ws-server/src/tests.rs

```diff
@@ -569,3 +587,58 @@ async fn run_forever() {
 	// Send the shutdown request from one handle and await the server on the second one.
 	join(server_handle.clone().stop().unwrap(), server_handle).with_timeout(TIMEOUT).await.unwrap();
 }
+
+#[tokio::test]
+async fn unsubscribe_twice_should_indicate_error() {
```

Comment:
ideally these tests should be in the RPC module but I have to refactor the `call_with` for that to work because inner `channel` is dropped in `call_with/call/execute`...